tools/build_config/build.ps1

```diff
@@ -1,29 +1,70 @@
+<#
+.SYNOPSIS
+  This script is used to build, test and package Herbert on Windows.
+
+.DESCRIPTION
+  This script requires Matlab, Visual Studio, CMake>=3.7 and CTest be installed
+  on your system and available on the path.
+
+  Use "Get-Help ./build.ps1 -Detailed" for parameter descriptions.
+
+.EXAMPLE
+  ./build.ps1 -build
+  # Builds Herbert
+.EXAMPLE
+  ./build.ps1 -test
+  # Runs all Herbert tests
+.EXAMPLE
+  ./build.ps1 -package
+  # Packages Herbert
+.EXAMPLE
+  ./build.ps1 -build -package -vs_version 2017 -matlab_release "R2019a"
+  # Builds and packages Herbert using Visual Studio 2017 and Matlab R2019a
+
+.LINK
+  https://github.com/pace-neutrons/Herbert
+#>
 param (
+  # Run the Herbert build commands
   [switch][Alias("b")]$build,
+  # Run all Herbert tests
   [switch][Alias("t")]$test,
+  # Pacakge Herbert into a .zip file
   [switch][Alias("p")]$package,
+  # Print the versions of libraries being used e.g. Matlab
   [switch][Alias("v")]$print_versions,
+  # Call Get-Help on this script and exit
+  [switch][Alias("h")]$help,
 
+  # The version of Visual Studio to build with (one of 2015, 2017, 2019).
+  # Other Windows compilers are not supported by this script. {2015, 2017, 2019}
```

Comment:
is there a standard way to show the default in the parameter help?

Reply:
You need to do `Get-help ./build.ps1 -Full` then the parameter output is a bit more verbose:
```powershell
    -build [<SwitchParameter>]
        Run the Herbert build commands.

        Required?                    false
        Position?                    named
        Default value                False
        Accept pipeline input?       false
        Accept wildcard characters?  false

    -test [<SwitchParameter>]
        Run all Herbert tests.

        Required?                    false
        Position?                    named
        Default value                False
        Accept pipeline input?       false
        Accept wildcard characters?  false
```

I'm not sure what you think is best here. Should I also include the default in the help string? It's more to manually maintain, but is more obvious from a user's perspective

Reply:
I think it would be worth it, that's a lot of extra clarity for a relatively small maintenance overhead and the default is defined in the same block of code a couple of lines below

Simple appending `... [ default-value ]` to the text would work e.g.
```
 # Run all Herbert tests [false]
[switch][Alias("t")]$test,
```
For enumerated options `..... {2015, 2017, 2019} [2017]` still parses reasonably clearly